src/HotChocolate/Data/src/EntityFramework/DataLoaders/EntityFrameworkLoader.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GreenDonut;
+using HotChocolate.DataLoader;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace HotChocolate.Data.DataLoaders
+{
+    public class EntityFrameworkLoader<TKey, TData, TProp, TDb> : BatchDataLoader<TData, ICollection<TProp>>
+        where TData : IModelId<TKey> where TProp : class, IModelId<TKey> where TDb : DbContext where TKey : class
+    {
+        private readonly IDbContextFactory<TDb> _dbContextFactory;
+
+        private List<ISkipNavigation>? _skipNavigations;
+
+        public EntityFrameworkLoader(
+            IBatchScheduler batchScheduler,
+            IDbContextFactory<TDb> dbContextFactory
+        ) : base(batchScheduler)
+        {
+            _dbContextFactory = dbContextFactory ?? throw new ArgumentNullException(nameof(dbContextFactory));
+        }
+
+        private List<ISkipNavigation> GetSkipNavigations(TDb context) =>
+            _skipNavigations ??= context.Model.GetEntityTypes(typeof(TData)).SelectMany(et => et.GetSkipNavigations()).ToList();
+
+        protected override async Task<IReadOnlyDictionary<TData, ICollection<TProp>>> LoadBatchAsync(
+            IReadOnlyList<TData> keys, CancellationToken cancellationToken
+        )
+        {
+            await using TDb db = _dbContextFactory.CreateDbContext();
+            List<TData> data = keys.ToList();
+
+            // TDB db = _dbFactory.Value;
+            List<ISkipNavigation> navigations = GetSkipNavigations(db);
+            ISkipNavigation? nav = navigations.FirstOrDefault(n => n.ClrType == typeof(ICollection<TProp>));
```

Comment:
It would be preferable to know the name of the attribute rather than relying on a single type. I could not find a way to pass this from the `desc.Field()` invocation into this resolver.